### PR TITLE
refactor: Refactor Collection to use the standalone components

### PR DESCRIPTION
**Pull request by rafenden**
Refactor the `CollectionList` component so it uses the newly created `Pagination`, `Badge` and `Metadata` components.

## Screenshot

localhost_54227_iframe html_id=collection--collection

### CODE REVIEW

**Comment by rafenden** [comment]
> Looks good. Think we just need to add a key for each `li` in the pagination rather than on the `a` - getting this error in the console:
> 
> ```
> vendors~main.e64d852ea6ddfc971380.bundle.js:118238 Warning: Each child in a list should have a unique "key" prop.
> 
> Check the render method of `Pagination`. See https://fb.me/react-warning-keys for more information.
>     in styled.li (created by Pagination)
>     in Pagination (created by CollectionList)
>     in CollectionList (created by CollectionWithState)
>     in CollectionWithState
> ```

Good point, code is now amended!

File: src/collection-list/CollectionItem.jsx [hunk 1]
```diff
@@ -1,61 +1,88 @@
...
-  ItemBadge,
-  ItemMetaWrapper,
-  ItemMeta,
-} from './items'
+import styled from 'styled-components'
+import { H3 } from '@govuk-react/heading'
+import Link from '@govuk-react/link'
+import { HEADING_SIZES, SPACING } from '@govuk-react/constants'
+import { GREY_2 } from 'govuk-colours'
+import Metadata from '../metadata/Metadata'
```
Comment (paulgain) [inline src/collection-list/CollectionItem.jsx]: Space after library imports.

File: src/collection-list/CollectionDownload.jsx [hunk 2]
```diff
@@ -2,39 +2,14 @@ import React from 'react'
...
-import {
-  SPACING,
-  MEDIA_QUERIES,
-  BORDER_WIDTH_FORM_ELEMENT,
-} from '@govuk-react/constants'
+import { MEDIA_QUERIES } from '@govuk-react/constants'
 import Button from '@govuk-react/button'
-import Paragraph from '@govuk-react/paragraph'
-import { BLACK } from 'govuk-colours'
+import CollectionHeaderRow from './CollectionHeaderRow'
```
Comment (paulgain) [inline src/collection-list/CollectionDownload.jsx]: Space after library imports :-)

File: src/collection-list/CollectionHeader.jsx [hunk 3]
```diff
@@ -5,62 +5,49 @@ import styled from 'styled-components'
 import pluralize from 'pluralize'
 import { H2 } from '@govuk-react/heading'
 import { BLACK, GREY_3 } from 'govuk-colours'
-import { SPACING, MEDIA_QUERIES, FONT_SIZE } from '@govuk-react/constants'
+import { HEADING_SIZES } from '@govuk-react/constants'
+import CollectionHeaderRow from './CollectionHeaderRow'
```
Comment (paulgain) [inline src/collection-list/CollectionHeader.jsx]: Space after library imports

File: src/collection-list/__tests__/CollectionDownload.test.jsx [hunk 4]
```diff
@@ -1,55 +1,47 @@
 import React from 'react'
 import { mount } from 'enzyme'
 import CollectionDownload from '../CollectionDownload'
```
Comment (paulgain) [inline src/collection-list/__tests__/CollectionDownload.test.jsx]: Space after library imports

File: src/collection-list/__tests__/CollectionHeader.test.jsx [hunk 5]
```diff
@@ -1,20 +1,14 @@
 import React from 'react'
 import { mount } from 'enzyme'
 import CollectionHeader from '../CollectionHeader'
```
Comment (paulgain) [inline src/collection-list/__tests__/CollectionHeader.test.jsx]: Space after library imports

File: src/collection-list/__stories__/CollectionList.stories.jsx [hunk 6]
```diff
@@ -1,114 +1,35 @@
...
-  interactionItem,
-  capitalProfileHeading,
-  paginationProps,
-  capitalProfileCollectionList1,
-} from '../__fixtures__'
-import CollectionItem from '../CollectionItem'
-import CollectionHeader from '../CollectionHeader'
-import CollectionDownload from '../CollectionDownload'
-import CollectionPagination from '../CollectionPagination'
 import CollectionList from '../CollectionList'
```
Comment (paulgain) [inline src/collection-list/__stories__/CollectionList.stories.jsx]: Space after library imports

File: src/collection-list/__tests__/CollectionDownload.test.jsx (hunk 4 above)
Comment (paulgain) [inline src/collection-list/__tests__/CollectionDownload.test.jsx]: Space after library imports

File: src/collection-list/__tests__/CollectionHeader.test.jsx (hunk 5 above)
Comment (paulgain) [inline src/collection-list/__tests__/CollectionHeader.test.jsx]: Space after library imports

File: src/collection-list/__tests__/CollectionList.test.jsx [hunk 7]
```diff
@@ -1,58 +1,66 @@
 import React from 'react'
 import { mount } from 'enzyme'
 import CollectionList from '../CollectionList'
```
Comment (paulgain) [inline src/collection-list/__tests__/CollectionList.test.jsx]: Space after library imports

**Comment by paulgain** [comment]
There should  be spacing between the badges.

spacing-between-badges

**Pull request merged**